File: 1-two-sum/two-sum.cpp
#include <vector>
#include <unordered_map>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::unordered_map<int, int> seen;
        for (int i = 0; i < nums.size(); ++i) {
            int complement = target - nums[i];
            if (seen.count(complement)) { 
                std::vector<int> result = {seen[complement], i};
                return result;
            }
            seen[nums[i]] = i; 
        }
        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        s= dict()
        sol = list([])

        for (idx,num) in enumerate(nums):
            if (target-num) in s:
                sol.append(s[target-num]);
                sol.append(idx)
                break
            s[num]=idx

        return sol